### PR TITLE
Added e2e tests for upgrading clusters with insufficient compute resources

### PR DESCRIPTION
**Pull request by mrog**
*Issue #, if available:*

*Description of changes:*
Added e2e tests for upgrading an existing cluster, with insufficient compute resources for the upgrade

*Testing performed:*
Ran e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by wongni — APPROVED** [review]

File: test/e2e/config/cloudstack.yaml [hunk 1]
```diff
@@ -125,6 +126,7 @@ variables:
   CLOUDSTACK_INVALID_CONTROL_PLANE_MACHINE_OFFERING: "OfferingXXXX"
   CLOUDSTACK_EXTREMELY_LARGE_CONTROL_PLANE_MACHINE_OFFERING: "Extremely Large Instance"
```
Comment (pmotyka) [inline test/e2e/config/cloudstack.yaml:127]: Should the offering name be more descriptive of the compute resource being modeled?

Comment (davidjumani) [reply test/e2e/config/cloudstack.yaml:129]: Would it be better to name it "impossible offering" ? It is misleading unless the user reads the test details

File: test/e2e/config/cloudstack.yaml [hunk 2]
```diff
@@ -127,6 +128,7 @@ variables:
...
   CLOUDSTACK_WORKER_MACHINE_OFFERING: "Medium Instance"
+  CLOUDSTACK_EXTREMELY_LARGE_WORKER_MACHINE_OFFERING: "Extremely Large Instance"
```
Comment (davidjumani) [inline test/e2e/config/cloudstack.yaml:131]: Here too

**Comment by davidjumani** [comment]
Apart from that comments, looks good, tests pass

File: test/e2e/config/cloudstack.yaml (hunk 1 above)
Comment (mrog) [reply test/e2e/config/cloudstack.yaml:129]: That's a good name for it.  I'll rename it.

Comment (mrog) [reply test/e2e/config/cloudstack.yaml]: That change is done and tested.

**Pull request merged**